python/contactflowhelper/lambda_function.py

```diff
@@ -0,0 +1,114 @@
+'''
+Basic helper function that covers common text/math operations for use with 
+contact flows. This function is assuming incoming events from a contact flow. 
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import json
+import random
+
+def lambda_handler(event, context):
+    response = {}
+    
+    if event['Details']['Parameters']['function'] == 'freeform_math':
+        response.update({'answer':str(freeform_math(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'random_number':
+        response.update({'answer':str(random_number(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'increment_1':
+        response.update({'answer':str(increment_1(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'increment_n':
+        response.update({'answer':str(increment_n(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'random_choice':
+        response.update({'answer':str(random_choice(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'replace_text':
+        response.update({'answer':str(replace_text(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'split_text':
+        response.update(split_text(event))
+        
+    elif event['Details']['Parameters']['function'] == 'strip_text':
+        response.update({'answer': str(strip_text(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'upper_text':
+        response.update({'answer': str(upper_text(event))})
+        
+    elif event['Details']['Parameters']['function'] == 'lower_text':
+        response.update({'answer': str(lower_text(event))})
+        
+    else:
+        response.update({'result':'fail'})
+        response.update({'reason':'no valid function passed'})
+    
+    response.update({'result':'success'})
```

Comment:
This overwrites the failed result. Move this before the if/else statements

Reply:
Fixed